mlfinlab/online_portfolio_selection/pattern_matching/functional_correlation_driven_nonparametric_learning.py

```diff
@@ -0,0 +1,110 @@
+# pylint: disable=missing-module-docstring
+import numpy as np
+import scipy.optimize as opt
+from mlfinlab.online_portfolio_selection.pattern_matching.correlation_driven_nonparametric_learning import \
+    CorrelationDrivenNonparametricLearning
+
+
+class FunctionalCorrelationDrivenNonparametricLearning(CorrelationDrivenNonparametricLearning):
+    """
+    This class implements the Functional Correlation Driven Nonparametric Learning strategy. It is
+    reproduced with modification from the following paper:
+    `Wang, Y., & Wang, D. (2019). Market Symmetry and Its Application to Pattern-Matching-Based
+    Portfolio Selection. The Journal of Financial Data Science, 1(2), 78–92.
+    <https://jfds.pm-research.com/content/1/2/78>`_
+
+    Functional Correlation Driven Nonparametric Learning is an extension of the SCORN strategy.
+    Unlike SCORN that sets a certain threshold for the objective function, FCORN places an
+    arbitrary weight to all relative returns period to emphasize certain weights. If the given
+    lambda value approaches infinity, then FCORN returns the same values as the SCORN strategy.
+    """
+
+    def __init__(self, window, rho, lambd):
+        """
+        Initializes Functional Correlation Driven Nonparametric Learning with the given window,
+        rho, and lambd value.
+
+        :param window: (int) Number of windows to look back for similarity sets.
+        :param rho: (float) Threshold for similarity.
+        :param lambd: (float) Scale factor for FCORN. As lambd approaches infinity,
+                              FCORN returns the same results as SCORN.
+        """
+        self.lambd = lambd
+        super().__init__(window=window, rho=rho)
```

Comment:
Consider describing possible values of lambd in more detail. Also, should the input values of lambd be tested? (Nonnegativity?)

Reply:
Lambd can actually take any value but loses its effectiveness if doing so. I'll put a description on that to clarify.